app/api/items/save_grade.go

```diff
@@ -86,22 +86,29 @@ func saveGradingResultsIntoDB(store *database.DataStore, user *database.User,
 	}
 
 	// Build query to update groups_attempts
-	// The score is set towards the end, so that the IF condition on
-	// best_answer_at is computed before score is updated
 	columnsToUpdate := []string{
 		"tasks_tried",
 		"latest_activity_at",
-		"best_answer_at",
+		"best_answer_id",
 		"latest_answer_at",
-		"score",
+		"score_computed",
 		"result_propagation_state",
 	}
 	values := []interface{}{
 		1,
 		database.Now(),
-		gorm.Expr("IF(? > score, ?, best_answer_at)", score, database.Now()),
+		// for best_answer_id we compare raw scores
```

Comment:
I am very sorry but your implementation has raised several questions on the "best answer" concept (in particular in case there is a bonus and the overall score is bound to 100)... and our conclusion is that we have to go backward and store the date of the first answer getting this score instead of the "best answer". 
Some of your work has to be reverted then, sorry about that. 😔

Let's forget about the "best_answer_id" and go back to storing "best_answer_at" but without the "best answer" naming which is quite confusing. Let's use "score_obtained_at" with the comment "Submission time of the first answer which led to the best score". 
For its computation, beware that if there is a "diff" of "+10" and the user has a submission S1 scoring 90 (so getting 10) and then another submission S2 score 100 (so getting 100 as well due to the max), it is the submission time of S1 which has to be stored as the score of the attempt is 100 and S1 is the first one to allow this score.

Reply:
done